pkg/deployments/index.ts

```diff
@@ -6,10 +6,13 @@ import { Artifact } from 'hardhat/types';
  * @param address Address of the contract to be fetched
  * @param network Name of the network looking the deployment for (e.g. mainnet,  polygon, goerli, etc)
  */
-export function lookupBalancerContractByAddress(address: string, network: string): { task: string; name: string } {
+export function lookupBalancerContractByAddress(
+  address: string,
+  network: string
+): { task: string; name: string } | null {
   // eslint-disable-next-line @typescript-eslint/no-var-requires
   const networkAddresses = require(getBalancerContractAddresses(network));
-  return networkAddresses[address];
+  return networkAddresses[address] ?? null;
```

Comment:
You think it's better to force the caller to handle this error than to simply throw an exception ourselves?